File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/providers/playlists/AppPlayList.java
package com.github.anrimian.musicplayer.data.storage.providers.playlists;

import androidx.annotation.NonNull;

import java.util.Date;

import javax.annotation.Nonnull;

public class AppPlayList {

    private final long id;

    private final long storageId;

    @Nonnull
    private final String name;

    @Nonnull
    private final Date dateAdded;

    @Nonnull
    private final Date dateModified;

    private final int compositionsCount;

    public AppPlayList(long id,
                       long storageId,
                       @Nonnull String name,
                       @Nonnull Date dateAdded,
                       @Nonnull Date dateModified,
                       int compositionsCount) {
        this.id = id;
        this.storageId = storageId;
        this.name = name;
        this.dateAdded = dateAdded;
        this.dateModified = dateModified;
        this.compositionsCount = compositionsCount;
    }

    public long getId() {
        return id;
    }

    public long getStorageId() {
        return storageId;
    }

    @Nonnull
    public String getName() {
        return name;
    }

    @Nonnull
    public Date getDateAdded() {
        return dateAdded;
    }

    @Nonnull
    public Date getDateModified() {
        return dateModified;
    }

    public int getCompositionsCount() {
        return compositionsCount;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AppPlayList that = (AppPlayList) o;

        return name.equals(that.name);
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }

    @NonNull
    @Override
    public String toString() {
        return "PlayList{" +
                "id=" + storageId +
                ", name='" + name + '\'' +
                ", dateAdded=" + dateAdded +
                ", dateModified=" + dateModified +
                '}';
    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/providers/playlists/AppPlaylist.kt
package com.github.anrimian.musicplayer.data.storage.providers.playlists

class AppPlaylist(
    val id: Long,
    val storageId: Long,
    val name: String,
    val addedTime: Long,
    val modifiedTime: Long,
    val compositionsCount: Int
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is AppPlaylist) return false

        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    override fun toString(): String {
        return "AppPlayList(id=$id, storageId=$storageId, name='$name', addedTime=$addedTime, modifiedTime=$modifiedTime, compositionsCount=$compositionsCount)"
    }

}
